builtin/providers/azurerm/resource_arm_security_group.go

```diff
@@ -0,0 +1,310 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmSecurityGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSecurityGroupCreate,
+		Read:   resourceArmSecurityGroupRead,
+		Update: resourceArmSecurityGroupCreate,
+		Delete: resourceArmSecurityGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"security_rule": &schema.Schema{
```

Comment:
Might the name `rule` here be more succinct and obvious? I can understand matching the API though.